### DISCUSSION TITLE
Where can I find more info about the geometric search system?

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#21622] [Q&A General] [answered]
When I read the introduction about the NearestNodeValueAux, it says"The NearestNodeValueAux uses the geometric search system to find the shortest distance from a node where the AuxVariable......".
I want to know more information about the geometric search system. And if possible, can anybody tell me how can I use this system?

Ethan

**Comment by GiudGiud** [comment, marked as answer]
Hello

We have this documentation online
https://mooseframework.inl.gov/moose/source/geomsearch/GeometricSearchData.html
Hope it helps
Guillaume

**Reply by Ethan-xj** [reply]
@GiudGiud Hello! It's an old disscusion :）
I want to know where I can find the info about the getQuadraturePenetrationLocator in [GeometricSearchData](https://mooseframework.inl.gov/source/geomsearch/GeometricSearchData.html). I want to express the coordinate of the quadrature points in Kernel.

**Reply by Ethan-xj** [reply]
Oh I find the GeometricSearchData.h and GeometricSearchData.C.
